azure-local/manage/upgrade-sdn.md

```diff
@@ -0,0 +1,406 @@
+    ```
+
+1. Check the OS after the upgrade:
+
+    ```powershell
+    # Check the OS version AFTER OS has been upgraded:
+    Get-ComputerInfo | Select-Object WindowsProductName, WindowsInstallationType, OSDisplayVersion, WindowsBuildLabEx | Format-Table -AutoSize
+    ```
+
+## Upgrade Hyper-V hosts
+
+> [!IMPORTANT]
+> If you deployed SDN on Azure Local, upgrade your Hyper-V hosts by following instructions in [About Azure Local upgrades](../upgrade/about-upgrades-23h2.md). Do not use the steps in this article for upgrading Hyper-V hosts.
+
+The upgrade process varies depending on the roles and services in your environment. If you have Storage Spaces Direct, clustering, or similar features, complete the necessary maintenance tasks to take a node offline for the upgrade process.
+
+If virtual machines (VMs) aren't using clustering, evaluate live migrating VMs to another host within the environment depending on capacity.
+
+After you put the Hyper-V host in the maintenance mode, follow the steps in [Perform in-place upgrade](#perform-in-place-upgrade). Repeat this process for all Hyper-V hosts within your cluster.
+
+## Upgrade Network Controller VMs (Service Fabric)
+
+Before you upgrade or restart a Network Controller VM, disable it in the Service Fabric cluster.
+
+Follow these steps on a remote computer that has WinRM connectivity to the Network Controller VMs.
+
+1. Retrieve the current state of the Network Controller nodes:
+
+    ```powershell
+    Get-SdnServiceFabricNode -NetworkController "<NC_VM>" | FT NodeName, IpAddressOrFQDN, NodeStatus, HealthState, IsStopped -AutoSize
+    ```
+
+1. Disable the node from the Service Fabric quorum. This step ensures Service Fabric can migrate primary replicas to other nodes and keep partition databases in sync.
+
+    ```powershell
+    Disable-SdnServiceFabricNode -NetworkController "<NC_VM>" -NodeName "<Node_Name>"
+    ```
+
+1. After the node is safely disabled, perform the in-place upgrade [Perform in-place upgrade](#perform-in-place-upgrade).
+
+1. After the node has completed the in-place upgrade, re-enable the node. This operation enables the node and waits until Service Fabric returns to a healthy state.
+
+    ```powershell
+    Enable-SdnServiceFabricNode -NetworkController "<NC_VM>" -NodeName "<Node_Name>"
+    ```
+
+    If the operation times out, wait and manually check the state:
+
+    ```powershell
+    Get-SdnServiceFabricNode -NetworkController "<NC_VM>" | FT NodeName, IpAddressOrFQDN, NodeStatus, HealthState, IsStopped -AutoSize
+
+    Confirm-SdnServiceFabricHealthy -NetworkController "<NC_VM>"
+    ```
+    
+      - If the commands return status healthy and node is up, repeat the process for other Network Controller nodes.
+        
+      - If the issue persists after an hour, see [Troubleshooting](#troubleshooting).
+
+1. Repeat the process for all Network Controller nodes in your deployment.
+
```

Comment:
For some weird reason I recall that the NC had 3 nodes? If so, should we say 3 instead of all?